### ISSUE TITLE
`node-param-default-missing`: false positive, and a duplicate `default` key inserted, when the existing default's value is not a literal

### ISSUE THREAD

**Issue by romerjon** [ivov/eslint-plugin-n8n-nodes-base#221]
`node-param-default-missing` reports a false positive — and its autofix **inserts a second `default` key** — when the existing `default`'s value is not a literal.

The rule appears to look for a `default` whose value it can resolve to a literal. When `default` is present but its value is a member expression, the rule behaves as though `default` were absent: it reports, and `--fix` inserts a **duplicate** `default` immediately after `type:`.

JavaScript keeps the **last** key, so the inserted one is usually inert and the damage is silent — `tsc` is happy and tests stay green. The real cost is the duplicate itself: the next reader deletes "the redundant one" with no way to tell which is live.

## Environment

| | |
|---|---|
| `eslint-plugin-n8n-nodes-base` | **1.16.5** |
| ESLint | **9.39.3** (flat config) |
| Node | 22.23.1 |
| `NODE_ENV` | **unset** (relevant to #197) |
| Parser | `@typescript-eslint/parser` |

Minimal flat config used for every reproduction below — the plugin is registered directly, with **one rule armed at a time**, so nothing else can be responsible:

```js
// eslint.config.mjs
import tsParser from "@typescript-eslint/parser";
import n8nNodesBase from "eslint-plugin-n8n-nodes-base";

export default [
  {
    files: ["**/*.ts"],
    languageOptions: { parser: tsParser, ecmaVersion: 2022, sourceType: "module" },
    plugins: { "n8n-nodes-base": n8nNodesBase },
    rules: { "n8n-nodes-base/RULE_NAME": "error" },
  },
];
```

The file under test is `nodes/Demo/Demo.node.ts`. Every snippet below is synthetic and self-contained — you can paste it and run it as-is.

I noticed #196 is still open on ESLint v9 support. These all reproduce on 9.39.3 with flat config; if there is a supported version matrix you would rather I verify against, tell me and I will re-run them.

## Reproduction

```ts
// nodes/Demo/Demo.node.ts
import type { INodeProperties } from "n8n-workflow";

const R = { ARTICLE: "article" } as const;

export const p: INodeProperties = {
  displayName: "Resource",
  name: "resource",
  type: "options",
  options: [{ name: "Article", value: "article" }],
  default: R.ARTICLE,
};
```

## Observed

```
5:33  error  Add a default [autofixable]  n8n-nodes-base/node-param-default-missing
```

`eslint --fix`:

```diff
 	displayName: 'Resource',
 	name: 'resource',
 	type: 'options',
+	default: 'article',
 	options: [{ name: 'Article', value: 'article' }],
 	default: R.ARTICLE,
```

Two `default` keys in one object literal.

## The trigger is specifically the `default` VALUE

I bisected this, because the position of the key turns out not to matter:

| `options[].value` | `default` value | rule fires |
|---|---|---|
| literal `"article"` | literal `"article"` | no |
| literal `"article"` | **`R.ARTICLE`** | **yes** |
| `R.ARTICLE` | literal `"article"` | no |

So it is not about `default` appearing late in the object, and not about the options entries — it is that the rule cannot resolve `default`'s value.

## Expected

Two things, independently:

1. Do not report when a `default` key is present, whatever its value resolves to. If the value cannot be resolved, that is not the same as the key being missing.
2. **Never insert a key that already exists at that level**, regardless of why the rule decided to report. Even if the report were correct, the fix should be a no-op rather than producing a duplicate.

The second is the important one: it makes this class of bug non-destructive even when the detection is wrong.

## Related

#222 is the same root cause in the other direction — `node-param-default-wrong-for-boolean` *rewrites* a `default` whose value is an identifier it cannot resolve, instead of leaving it alone. A shared "if the initialiser is not a resolvable literal, report nothing and change nothing" policy across the `default`-related rules would close both.
